crates/kornia-pnp/src/ransac.rs

```
@@ -0,0 +1,492 @@
+        }
+
+        // Score model on all points
+        let (inliers, _) = classify_inliers(
+            world,
+            image,
+            &pose_min.rotation,
+            &pose_min.translation,
+            &intr_x,
+            &intr_y,
+            params.reproj_threshold_px,
+        );
+
+        if inliers.len() > best_inliers.len() {
+            best_inliers = inliers;
+            best_pose = Some(pose_min.clone());
+
+            // Update required iterations based on current inlier ratio and sample size
+            if best_inliers.len() >= sample_size {
+                let w = best_inliers.len() as f32 / n as f32;
+                let s = sample_size as f32;
+
+                // Avoid numerical issues with very small w
+                if w > 1e-6 && w < 1.0 {
+                    let ws = w.powf(s);
+                    if ws < 1.0 - 1e-12 && ws > 1e-12 {
+                        // Avoid log(0) and log(1)
+                        let log_conf = (1.0 - params.confidence).max(1e-12).ln();
+                        let log_denom = (1.0 - ws).ln();
+                        if log_denom.is_finite() && log_denom != 0.0 {
+                            let est = (log_conf / log_denom).ceil();
+
+                            if est.is_finite() && est > 0.0 {
+                                let est_usize = est.min(params.max_iterations as f32) as usize;
+                                if est_usize < required_iters {
+                                    required_iters = est_usize;
+                                }
+                            }
+                        }
+                    } else if w >= 0.95 {
+                        // Very high inlier ratio (≥95%), we can stop early
+                        required_iters = iter;
+                    }
+                }
+            }
+        }
+    }
+
+    // Validate and optionally refine
+    if best_inliers.len() < 4 {
+        return Err(PnPError::InsufficientInliers {
```

do we want to return an error here ? to me it's a None